app/controllers/katello/api/v2/content_export_incrementals_controller.rb

```diff
@@ -0,0 +1,99 @@
+    param :chunk_size_mb, :number, :desc => N_("Chunk export-tarfile into pieces of chunk_size mega bytes."), :required => false
+    param :from_history_id, :number, :desc => N_("Export history identifier used for incremental export. "\
+                                                 "If not provided the most recent export history will be used."), :required => false
+    def library
+      tasks = async_task(::Actions::Pulp3::Orchestration::ContentViewVersion::ExportLibrary,
+                          @organization,
+                          destination_server: params[:destination_server],
+                          chunk_size: params[:chunk_size_mb],
+                          from_history: @history)
+      respond_for_async :resource => tasks
+    end
+
+    private
+
+    def find_exportable_content_view_version
+      @version = ContentViewVersion.exportable.find_by_id(params[:id])
+      throw_resource_not_found(name: 'content view version', id: params[:id]) if @version.blank?
+      @view = @version.content_view
+    end
+
+    def find_library_export_view
+      @view = ::Katello::ContentView.find_library_export_view(destination_server: params[:destination_server],
+                                                                organization: @organization,
+                                                                create_by_default: false)
+      if @view.blank?
+        msg = _("Unable to incrementally export. Do a Full Export the library content "\
+                "before updating from the latest increment.")
+        fail HttpErrors::BadRequest, msg
+      end
+    end
+
+    def find_history
+      if !params[:from_history_id].blank?
+        @history = ::Katello::ContentViewVersionExportHistory.find(params[:from_history_id])
+        if @history.blank?
+          throw_resource_not_found(name: 'content view version export history',
+                                   id: params[:from_history_id])
+        end
+      else
+        @history = ::Katello::ContentViewVersionExportHistory.
+                      pick_recent_history(@view,
+                                          destination_server: params[:destination_server])
+        if @history.blank?
+          if params[:action] == :library
+            msg = _("Prior content view export history belonging to the library export content view was not found "\
+                    "in organization with id %{id} and destination_server %{server}. "\
+                    "Please choose the full export.") %
+                    {id: @organization.id, server: params[:destination_server]}
+          else
+            msg = _("Prior content view export history belonging to the content view '%{view}' was not found "\
+                    " for the destination_server '%{server}'. Please choose the full export.") %
+                    {view: @view.name, server: params[:destination_server]}
```

Comment:
How strongly do you feel about having these two distinct error messages? In other words how much do they really help the user? What do you think about `No existing export history was found to perform an incremental export. A full export must be performed`

Reply:
Not very strongly :) . Will change it.